### ISSUE TITLE
[Design, FE] Notification Prefs: Begin compiling/executing accessibility tests for staging review

### ISSUE THREAD

**Issue by ajakabcin** [department-of-veterans-affairs/va.gov-team#30341]
## Background
As part of our preparation for the staging review we need to perform accessibility tests for Notification Preferences. This will be a joint effort from Design and FE to perform these tests and Product to ensure everything is documented to submit the request. The plan is for @ajakabcin to be able to submit this request in sprint 58.

[A11y staging review preparation](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/staging-review-processes.md#accessibility-staging-review-preparation) documentation for reference. 

## Tasks
Design and FE to perform the accessibility tests as documented [here](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/identity-personalization/my-va/2.0-redesign/staging-review/accessibility-test-prep.md)

**Comment by andaleliz** [comment]
@joshkimux or @AngelaFowler82, I'm going through the keyboard navigation tests for this and ran into something I don't quite understand on the radio buttons. I tested a page that had two groups of radio inputs. Each group has two radio options, so there are a total of 4 radio buttons on the page.  

When I'm tabbing down the page, the first radio input in the first group gets focus.  Then if I press tab again, the second input in the group is skipped, the whole second group is skipped, and the focus goes down into the footer.  That doesn't seem right.  What do you think?

**Comment by AngelaFowler82** [comment]
You're right, that is strange. What test account are we using? I thought it was user+36, but it needs a phone number and won't let me add one.

**Comment by joshkimux** [comment]
I think I know why. These radios might _all_ be sharing the same id pairing, so they all might be rendering as the same group.

**Comment by joshkimux** [comment]
It's normal behavior for `tab` to set focus to the first radio option, and then the second `tab` to go to the next interactive element (not the second radio option, which is meant to be reached with the arrow keys)

**Comment by erikphansen** [comment]
I'm seeing slightly different, but similar behavior with tabbing and am scratching my head. Focus goes to the first button group/fieldset. Hitting `tab` skips the next group/fieldset and sets focus on the link between the second and third radio button groups. The next `tab` puts focus on the third group/fieldset. It's treating the first four radio buttons as though they are all one group. But each pair of radio buttons is in its own fieldset with unique IDs, so I don't understand why it'd skip one of them.

https://user-images.githubusercontent.com/20728956/135190528-668f7a8b-b807-45ba-b1b0-9b3572377327.mp4

**Comment by joshkimux** [comment]
@erikphansen , sorry I got my stuff mixed up-- not the IDs, in this case it would be the `name` attribute that groups them.

> A radio group is defined by giving each of radio buttons in the group the same name. Once a radio group is established, selecting any radio button in that group automatically deselects any currently-selected radio button in the same group.

[From MDN on radio groups](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio)

**Comment by erikphansen** [comment]
Thank you @joshkimux ! Found it and this should be an easy adjustment to make 🎉

**Comment by andaleliz** [comment]
@joshkimux and @AngelaFowler82  one more question - when I use the arrow keys to go to the next radio button in a group, it automatically selects it for me.  Wouldn't the user want to know what the option is before selecting it? What is the expected way a user would navigate the radio options to understand what is available before making a selection?

**Comment by joshkimux** [comment]
More than happy to answer these 😄 
1. This is completely standard behavior, and screen reader users will be used to this. Having a cancel button helps here. Alternatively, you may want to consider a radio button for "neither" or "no choice"
2. Hmm, that's odd. If you tab to the first radio button, it should announce it's legend and placement. If there's no legend, that's probably the issue! Otherwise, double check you're using Safari instead of Chrome for your browser. I can check the code out first thing tomorrow if needed 😄

**Comment by erikphansen** [comment]
Angela definitely tested item 2 and it worked properly. It read the aria-label for the given radio button label as it should. And it worked for me as well when I tested it. But I'll take another look myself.

**Comment by erikphansen** [comment]
@andaleliz Angela specifically requested that the legends be changed to h3's for easier navigation. And the legends didn't function as legends for screen readers because of the aria-label for each radio button option.

**Comment by andaleliz** [comment]
Okay, sounds good @erikphansen. I won't create any bug tickets then. She is definitely a more experienced screen reader user than I am.

**Comment by erikphansen** [comment]
@andaleliz When you are testing with macOS Voiceover, which browser are you using? Also, I had the same concerns as you re: using arrows to move between radio buttons which also _selects_ them. But if you use VO+→ to jump to the next radio button or radio button label. Which basically accomplishes the goal.

**Comment by andaleliz** [comment]
Yep, I used Safari + Chrome to test VO. @erikphansen are you up for doing Firefox + NVDA to complete that row in our table?

I identified a few possible problems that I'm going to bring to office hours tomorrow.  I'll create issues related to this if Angela agrees they're problematic. 
- VO reads out things like "black square" and "horizontal dividing line". Is that extraneous stuff we should hide from screen readers?
- When navigating via the keyboard, when I hit the `select your notification options` link in the alert, I'm sent down the page (expected) but I don't see where the focus is going.  I _think_ it should go to the first radio button, but it's actually going to the  `h3`. Not totally sure which is correct.
- When I add a phone number, VO doesn't tell me that the text `manage your notification settings` in the alert is a link

**Comment by joshkimux** [comment]
@andaleliz I can answer some of these ahead of time!
- black square is perfectly normal in lists, no need to be concerned about that
- horizontal dividing line is the screen reader announcing out the `hr`; if the line is an important visual indicator that a new section is beginning, it should announce (otherwise you can hide it with `aria-hidden="true"`)
- if sent to the heading, not showing the focus halo is consistent with VA design system standards because non-interactive elements (like a `h3` for example) shouldn't look like they can be interacted with; as for whether or not it should go to the first radio button is a good question for Angela!

**Comment by erikphansen** [comment]
1. I'll hide the `hr` from screen readers
2. In office hours with Angela a week ago we tested the `select your notification options` link behavior and I specifically asked her if focus was managed correctly in this scenario and it totally made sense to her. But I think that is when she asked that the `legends` be converted to `h3`s.
3. Good point, I should manage focus better. In my tests the alert was _read_ after successfully adding the phone number but it wouldn't be clear to a screen reader user that there was a link to click or how to get to that link.

**Comment by andaleliz** [comment]
@erikphansen all of that sounds good. 

> In my tests the alert was read after successfully adding the phone number 

I had the same results.

**Comment by andaleliz** [comment]
We reviewed with Josh and Angela in office hours, here are the outcomes:
1. We're good to proceed with hiding the `hr` from screenreaders
2. The `select your notification options` link behavior needs some tweaking.  Our a11y folks recommend using both [heading AND legend together](https://design-system.service.gov.uk/get-started/labels-legends-headings/), and also improving the focus management ([created this ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/30812)). In our meeting we discussed focus should go to the h3 but I realized that won't work once we have multiple notifications available in a group and a user has some set but not others.  I'm recommending it go to the label so SR will announce that to give context to users.

**Comment by ajakabcin** [comment]
Thank you for the thorough work here @andaleliz !

**Comment by erikphansen** [comment]
@ajakabcin might as well. I already have a PR up for that https://github.com/department-of-veterans-affairs/vets-website/pull/18992
